contracts/src/spaces/facets/tipping/TippingFacet.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+// interfaces
+import {ITipping} from "./ITipping.sol";
+
+// libraries
+import {TippingBase} from "./TippingBase.sol";
+
+// contracts
+import {ERC721ABase} from "contracts/src/diamond/facets/token/ERC721A/ERC721ABase.sol";
+import {Facet} from "contracts/src/diamond/facets/Facet.sol";
+
+contract TippingFacet is ITipping, ERC721ABase, Facet {
+  function __Tipping_init() external onlyInitializing {
+    _addInterface(type(ITipping).interfaceId);
+  }
+
+  /// @inheritdoc ITipping
+  function tip(TipRequest calldata tipRequest) external payable {
+    address sender = msg.sender;
```

Comment:
[`msg.sender`](https://www.evm.codes/?fork=cancun#a0) costs 2 gas which is cheaper than pushing it to the stack and moving it on the stack.